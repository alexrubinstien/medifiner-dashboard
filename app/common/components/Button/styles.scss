@import './app/common/styles/utils/_vars.scss';

.button {
  display: inline-block;
  width: 220px;
  padding: 10px 8px;
  background: $primary-color;
  border: 0;
  border-radius: 3px;
  color: #fff;
  font-size: 16px;
  font-weight: 700;
  line-height: 26px;
  letter-spacing: 1.14px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  transition: all 0.25s ease;

  &:hover {
    background: $primary-color-hover;
  }

  &--secondary {
    padding: 8px;
    background: transparent;
    border: 2px solid $border-color;
    color: $primary-color;
    font-size: 14px;
    letter-spacing: 1px;

    &:hover {
      background: transparent;
      border-color: $primary-color-hover;
      color: $primary-color-hover;
    }
  }

  &--wide {
    width: 100%;
  }
}