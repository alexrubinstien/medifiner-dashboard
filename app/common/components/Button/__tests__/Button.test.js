/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../index.js';

describe('With Snapshot Testing', () => {
  it('Button component', () => {
    const component = renderer.create(
      <Button>
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button component with custom class', () => {
    const component = renderer.create(
      <Button className="test-class">
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Secondary button component', () => {
    const component = renderer.create(
      <Button secondary>
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Link as button component', () => {
    const component = renderer.create(
      <Button
        isLink
        href="/"
        as="/"
      >
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Link as button component with custom class', () => {
    const component = renderer.create(
      <Button
        isLink
        href="/"
        as="/"
        className="test-class"
      >
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Link as secondary button component', () => {
    const component = renderer.create(
      <Button
        secondary
        isLink
        href="/"
        as="/"
      >
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Wide button component', () => {
    const component = renderer.create(
      <Button wide>
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Wide secondary button component', () => {
    const component = renderer.create(
      <Button
        secondary
        wide
      >
        Button
      </Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
