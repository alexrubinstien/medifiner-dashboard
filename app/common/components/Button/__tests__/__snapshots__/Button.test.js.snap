// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Button component 1`] = `
<button
  className="button"
  type="button"
>
  Button
</button>
`;

exports[`With Snapshot Testing Button component with custom class 1`] = `
<button
  className="test-class button"
  type="button"
>
  Button
</button>
`;

exports[`With Snapshot Testing Link as button component 1`] = `
<a
  className="button"
  href="/"
  onClick={[Function]}
>
  Button
</a>
`;

exports[`With Snapshot Testing Link as button component with custom class 1`] = `
<a
  className="test-class button"
  href="/"
  onClick={[Function]}
>
  Button
</a>
`;

exports[`With Snapshot Testing Link as secondary button component 1`] = `
<a
  className="button button--secondary"
  href="/"
  onClick={[Function]}
>
  Button
</a>
`;

exports[`With Snapshot Testing Secondary button component 1`] = `
<button
  className="button button--secondary"
  type="button"
>
  Button
</button>
`;

exports[`With Snapshot Testing Wide button component 1`] = `
<button
  className="button button--wide"
  type="button"
>
  Button
</button>
`;

exports[`With Snapshot Testing Wide secondary button component 1`] = `
<button
  className="button button--secondary button--wide"
  type="button"
>
  Button
</button>
`;
